### PR TITLE
[v7r3] Use full jobID in AREX jobsToCancel

### PR DESCRIPTION
**Pull request by sfayer**
Hi,

I've noticed that the AREX jobsToCancel list was being built out of ARC IDs rather than full DIRAC job IDs... I think the killJob function needs the latter. I've not seen this happen, but it appears it'll just result in a warning being logged for each ID, but the fewer spurious warnings we generate the better!

Regards,
Simon

BEGINRELEASENOTES
*Resources
FIX: Use full jobID in AREX jobsToCancel
ENDRELEASENOTES

### CODE REVIEW

**Review by fstagni — DISMISSED** [review]

**Comment by aldbr** [comment]
Indeed, `killJob()` is expecting a DIRAC job ID, but since we already have the ARC job ID, we can avoid one operation.
`DIRACToArcID()` will just return the ARC job ID passed in parameter.
But ok anyway this is minor.

Maybe we could also change the level of the logs in `DiracToArcID()` and `arcToDiracID()`: from `warn` to `verbose` or `debug`.
What do you think?

**Comment by sfayer** [comment]
I agree that doing the conversion multiple times is a bit excessive... Perhaps we could split killJob into an internal _killJob function that uses the ArcIDs, call that from getJobStatus and in the public killJob function just do the DiracToArcID and then call _killJob. That way we'd only do the conversion once where needed and avoid spurious logging, what do you think?

**Comment by aldbr** [comment]
Alright, yes I agree.
Do you want to take care of this?

**Comment by sfayer** [comment]
Sure, done.

File: src/DIRAC/Resources/Computing/AREXComputingElement.py [hunk 1]
```diff
@@ -393,38 +429,11 @@ def killJob(self, jobIDList):
 
         :param list: list of DIRAC Job IDs
```
Comment (aldbr) [inline src/DIRAC/Resources/Computing/AREXComputingElement.py]:
```suggestion
        :param list jobIDList: list of DIRAC Job IDs
```

File: src/DIRAC/Resources/Computing/AREXComputingElement.py [hunk 2]
```diff
@@ -321,6 +321,42 @@ def _uploadJobDependencies(self, arcJobID, executableFile, inputs, executables):
             self.log.verbose("Input correctly uploaded", fileToSubmit)
         return S_OK()
 
+    def _killJob(self, jobIDList):
+        """Kill the specified jobs
+
+        :param list: list of ARC Job IDs
```
Comment (aldbr) [inline src/DIRAC/Resources/Computing/AREXComputingElement.py]:
```suggestion
        :param list arcJobList: list of ARC Job IDs
```

File: src/DIRAC/Resources/Computing/AREXComputingElement.py [hunk 3]
```diff
@@ -321,6 +321,42 @@ def _uploadJobDependencies(self, arcJobID, executableFile, inputs, executables):
             self.log.verbose("Input correctly uploaded", fileToSubmit)
         return S_OK()
 
+    def _killJob(self, jobIDList):
```
Comment (aldbr) [inline src/DIRAC/Resources/Computing/AREXComputingElement.py]: Just to be in line with `renewJobs(self, arcJobList)`:
```suggestion
    def _killJob(self, arcJobList):
```

File: src/DIRAC/Resources/Computing/AREXComputingElement.py [hunk 4]
```diff
@@ -321,6 +321,42 @@ def _uploadJobDependencies(self, arcJobID, executableFile, inputs, executables):
...
+
+        # Get a proxy
+        result = self._prepareProxy()
+        if not result["OK"]:
+            self.log.error("Failed to set up proxy", result["Message"])
+            return result
+        self.session.cert = Locations.getProxyLocation()
+
+        # List of jobs in json format for the REST query
+        jobsJson = {"job": [{"id": job} for job in jobIDList]}
```
Comment (aldbr) [inline src/DIRAC/Resources/Computing/AREXComputingElement.py]: Oops, it should be changed here too. This should fix the problems with the CI: 
```suggestion
        jobsJson = {"job": [{"id": job} for job in arcJobList]}
```

**Review by aldbr — APPROVED** [review]

**Review by fstagni — APPROVED** [review]

**Pull request merged**